### PR TITLE
Edge macro readme

### PR DESCRIPTION
**Pull request by bojanserafimov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1296,6 +1299,146 @@ CREATE VIEW animal AS
...
+{
+    Animal {
+        name @output(out_name: "animal_name")
+        out_Animal_GrandparentOf {
+            name @output(out_name: "grandchild_name")
+        }
+    }
+}
+```
+Macro edges can make use of tags, filters, etc, to define more complicated macros like this one
```
Comment (pmantica1) [inline README.md]: Remove the multiple filters and tags. Doesn't make the example any more educational.  I would prefer if you showed me how using macros with tags work and how tag name resolution works. Feel free to add this in your end-to-end example below.

File: README.md [hunk 2]
```diff
@@ -1296,6 +1299,146 @@ CREATE VIEW animal AS
...
+            name @output(out_name: "animal_name")
+            out_Animal_GrandparentOf {
+                name @output(out_name: "grandchild")
+            }
+        }
+    }'''
+    args = {}
+    expanded_query, new_args = macros.perform_macro_expansion(macro_registry, query, args)
+    
+    # Verify
```
Comment (pmantica1) [inline README.md]: Verify what?

File: README.md [hunk 3]
```diff
@@ -1296,6 +1299,146 @@ CREATE VIEW animal AS
...
+    query = '''{
+        Animal {
+            name @output(out_name: "animal_name")
+            out_Animal_GrandparentOf {
+                name @output(out_name: "grandchild")
+            }
+        }
+    }'''
+    args = {}
+    expanded_query, new_args = macros.perform_macro_expansion(macro_registry, query, args)
```
[the remark this answers is not in the document]
Comment (bojanserafimov) [reply README.md]: This is a demo of the macro registry only

File: README.md (hunk 1 above)
Comment (bojanserafimov) [reply README.md]: Why do you care about name resolution? That will just clutter the main point of macros.

Comment (pmantica1) [reply README.md]: I am sorry my bad when you mentioned tags for some reason I was thinking of runtime parameters instead.

File: README.md [hunk 4]
```diff
@@ -1296,6 +1299,146 @@ CREATE VIEW animal AS
...
+Macro definitions and macro uses are type checked. If a macro is invalid, or is used incorrectly,
+the macro system will complain, rather than create an invalid query.
+
+#### Constraints for macro definitions
+- Macro definitions cannot use other macros as part of their definition.
+- A macro requires exactly one `@macro_edge_definition` and one `@macro_edge_target` directive.
+Additionally, these directives can only be used inside macro definitions.
+- The `@macro_edge_target` cannot be inside a `@fold` scope.
+- The `@macro_edge_target` cannot begin with a type coercion. If that is what you want, put
+the `@macro_edge_target` on the type coercion itself, instead of before it.
```
Comment (pmantica1) [inline README.md]: Can you a macro include runtime parameters?

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (pmantica1) [reply README.md]: Ok but at least add a comment that links to a piece of text or example of how args are affected. I think that anybody who reads through this piece of code will have that question.

[the remark this answers is not in the document]
Comment (bojanserafimov) [reply README.md]: ok makes sense

File: README.md [hunk 5]
```diff
@@ -1296,6 +1299,156 @@ CREATE VIEW animal AS
...
+            name @output(out_name: "animal_name")
+            out_Animal_GrandchildrenCalledNate {
+                uuid @output(out_name: "grandchild")
+            }
+        }
+    }'''
+    args = {}
+    expanded_query, new_args = macros.perform_macro_expansion(macro_registry, query, args)
+    
+    # Verify that macro expansion worked
```
Comment (pmantica1) [inline README.md]: Remove this line.

File: README.md [hunk 6]
```diff
@@ -1296,6 +1299,156 @@ CREATE VIEW animal AS
...
+            net_worth @filter(op_name: ">", value: ["%net_worth"])
+            out_Animal_BornAt {
+                event_date @filter(op_name: "<", value: ["%birthday"])
+            }
+        }
+    }
+}
+```
+Macro edges can also use runtime parameters. The values for the parameters are provided when
+the macro is registered, and not when it is used. See the section below to see how to use
```
Comment (pmantica1) [inline README.md]: See [Using the macro registry](#using-the-macro-registry) for an example of how to use macros with runtime parameters.

**Pull request merged**